primer/src/Primer/Action/Available.hs

```diff
@@ -397,13 +403,41 @@ forTypeDefParamNode paramName l Editable tydefs defs tdName td =
         )
         [NoInput DeleteTypeParam]
 
+forTypeDefParamKindNode ::
+  TyVarName ->
+  ID ->
+  Level ->
+  Editable ->
+  TypeDefMap ->
+  DefMap ->
+  TyConName ->
+  ASTTypeDef TypeMeta KindMeta ->
+  [Action]
+forTypeDefParamKindNode _ _ _ NonEditable _ _ _ _ = mempty
+forTypeDefParamKindNode paramName id l Editable tydefs defs tdName td =
+  sortByPriority
+    l
+    $ mwhen (not $ typeInUse tdName td tydefs defs)
+    $ [NoInput MakeKFun] <> case findKind id . snd =<< find ((== paramName) . fst) (astTypeDefParameters td) of
+      Nothing -> []
+      Just (KHole _) -> [NoInput MakeKType]
+      Just _ -> [NoInput DeleteKind]
+  where
+    findKind i k =
+      if getID k == i
+        then Just k
+        else case k of
+          KHole _ -> Nothing
+          KType _ -> Nothing
+          KFun _ k1 k2 -> findKind i k1 <|> findKind i k2
```

Comment:
This is another thing which can be simplified once we have zippers etc. for kinds.